### DISCUSSION TITLE
Geode Mod Packs

### DISCUSSION THREAD

**Discussion by TechStudent10** [geode-sdk/sdk#72] [Geode Ideas]
Kind of like in some MC launchers, you can change which mods get loaded through mod packs.

This would help by loading mods you don't need at the moment, therefore speeding up GD. 

I'm not sure if Geode already has a way for sort of dynamically loading mods, but if not, this could a great idea.

Example, if you are creating levels, enable the creator pack. 

Beating list demons? Enable the Demon pack. 

I know that Geode already allows you to toggle mods off and on, and this would essentially enable and disable a bunch of mods at once. 

What do you guys think?

**Comment by camila314** [comment]
all mods are loaded at startup or on install, so i don't think there would be a performance benefit

**Reply by TechStudent10** [reply]
Ah alright. I still think that being able to enable and disable multiple mods at a time with a single click is still in advantage.

**Reply by HJfod** [reply]
There could be a performance benefit, as for example EditorCollab by alk takes a lot of time to load (around 2-3 seconds). Right now that load time isn't noticable as the mods are loaded simultaniously on different threads, but it is still processing and memory power wasted if the mod isn't used in that session.

Mod packs are definitely something I want to look into, although I think they would likely be exclusively an index thing (i.e. only for installing a bunch of mods together). For picking which mods to load at startup, I think some kind of profile system would be better, although we might have to make some sort of launcher for that (or make it so picking a different profile restarts the game). I'll look into that.

**Reply by TechStudent10** [reply]
> There could be a performance benefit, as for example EditorCollab by alk takes a lot of time to load (around 2-3 seconds). Right now that load time isn't noticable as the mods are loaded simultaniously on different threads, but it is still processing and memory power wasted if the mod isn't used in that session.
> 
> Mod packs are definitely something I want to look into, although I think they would likely be exclusively an index thing (i.e. only for installing a bunch of mods together). For picking which mods to load at startup, I think some kind of profile system would be better, although we might have to make some sort of launcher for that (or make it so picking a different profile restarts the game). I'll look into that.

When I originally thought of this idea, I thought about making a mod pack/launcher type deal, but then I started my personal project and focus shifted there.
